tgp-2prpl.c

```diff
@@ -193,6 +197,88 @@ int p2tgl_imgstore_add_with_id_raw (const unsigned char *raw_bgra, unsigned widt
   return purple_imgstore_add_with_id (tga, tga_len, NULL);
 }
 
+#ifdef HAVE_LIBPNG
+
+struct mem_encode
```

Comment:
This kinda looks like the same struct as a GByteArray, would you consider using that instead so you can then take advantage of g_bytearray_* funcs instead of manual memory manipulations?